app/controllers/submissions_controller.rb

```diff
@@ -9,6 +9,7 @@ def create
 
     @task = Task.find(params[:submission][:task_id])
     @task.update status: :waiting if @task.not_submitted? || @task.waiting?
+    @task.update status: :new_version if @task.accepted_partially?
```

Comment:
I'll change names to `new_submission`, `new_submission_with_notes`. But I'm not sure it's worth to change `accepted_partially` since it describes a task's status well.